### PR TITLE
Feature/creating pagination

### PR DESCRIPTION
**Pull request by Demavend**

### CODE REVIEW

File: src/components/forms/pagination/pagination-form.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="btn-group">
+    <button data-id="prev" class="btn btn-default">Prev</button>
```
Comment (the-daria) [inline src/components/forms/pagination/pagination-form.html]: Please, remove prev, next buttons.

File: src/app.js [hunk 2]
```diff
@@ -1,10 +1,29 @@
 /**
+ * @member {Object} books - declaration the new instance of collection Books
+ * which contains the last 10 books from the query
  * @member {Object} library - declaration the new instance of collection
+ * which contains all books from the query
  */
 Collections = {
-    books: new Books()
+    books: new Books(),
+    library: new Library()
```
Comment (the-daria) [inline src/app.js]: For what do you need library collection ?

File: src/app.js [hunk 3]
```diff
@@ -37,18 +56,31 @@ const MainRouter = Backbone.Router.extend(
          * @param {Number} page - number of the page
          */
         search: (str, page) => {
+            /*if (page > 1 && Collections.library.length == 0) {
```
Comment (the-daria) [inline src/app.js]: Remove comments from PR, please.

File: src/app.js [hunk 4]
```diff
@@ -37,18 +56,31 @@ const MainRouter = Backbone.Router.extend(
...
-
 /**
- * Method that creates a new Router and start history
+ * Method that creates a new Router, start history
+ * and listens to collections for changes
  */
 $(() => {
     const app = new MainRouter();
     Backbone.history.start();
+    Collections.books.on('sync', () => {
```
Comment (the-daria) [inline src/app.js]: It should be done inside of the component.

File: src/app.js [hunk 5]
```diff
@@ -37,18 +56,31 @@ const MainRouter = Backbone.Router.extend(
...
+        let lib = Collections.books.toJSON();
+        $('.pagination').before(new List(lib).render().el);
+        Collections.library.add(Collections.books.models);
+    });
+    Collections.library.on('update', function(e) {
```
Comment (the-daria) [inline src/app.js]: It is incorrect. Please remove this part from here.

File: src/components/forms/pagination/pagination-form.view.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
+const PaginationForm = Backbone.View.extend(
```
Comment (the-daria) [inline src/components/forms/pagination/pagination-form.view.js]: This is not a form. Please rename it and move in the root of components folder.

File: src/components/forms/pagination/pagination-form.view.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            Backbone.history.navigate(`search/${this.params.query}/${this.params.page}`);
+            this.removePanels();
+            this.checkData((this.params.page * STEP) - STEP);
+        },
+        /**
+         * Method that adds 10 following books
+         * @fires SearchPage#showMore
+         */
+        showMore: function() {
+            Backbone.history.navigate(`search/${this.params.query}/${this.params.page + 1}`);
```
Comment (the-daria) [inline src/components/forms/pagination/pagination-form.view.js]: You don't need to increment page twice. Just do ++this.params.page before navigate.

File: src/components/list/list-item.view.js [hunk 8]
```diff
@@ -8,7 +8,7 @@ const ListItem = Backbone.View.extend(
          * @member {String} className - the class attribute of the element
```
Comment (the-daria) [inline src/components/list/list-item.view.js]: list-item.html is missed.

File: src/app.js (hunk 2 above)
Comment (Demavend) [reply src/app.js]: The librari collection is needed to store earlier requests.

Comment (Demavend) [reply src/app.js]: Librari collection is needed to store earlier requests

File: src/components/list/list-item.view.js (hunk 8 above)
Comment (Demavend) [reply src/components/list/list-item.view.js]: list-item.html is located in src/components/list/list-item.html. I do not observe any problems with this.

File: src/pages/search-page/search.view.js [hunk 9]
```diff
@@ -10,42 +10,75 @@ const SearchPage = Backbone.View.extend(
...
+        },
+        /**
+         * Method that show modal window
+         * @param {Object} e - get value of current event
+         * @member {Object} target - determines which
+         * collection object matches the event
+         */
+        showModal: function(e) {
+            let target = Collections.library.toJSON().find((item) => {
+                return item.id === e.target.getAttribute('data-id');
```
Comment (the-daria) [inline src/pages/search-page/search.view.js]: use short es6 syntacsis item => item.id === e.target.getAttribute('data-id')

File: src/pages/search-page/search.view.js [hunk 10]
```diff
@@ -10,42 +10,75 @@ const SearchPage = Backbone.View.extend(
...
          * Creates a new SearchPage instance
          * @constructs
          * @extends Backbone.View
-         * @param {{}} params - Backbone.View options
+         * @param {Object.<string, number>} params - landmark for search and page number
          */
         initialize: function(params) {
             this.params = params;
+            Collections.books.on('sync', () => {
+                console.log(1);
```
Comment (the-daria) [inline src/pages/search-page/search.view.js]: Please, remove it.

**Pull request merged**